metadata/src/main/java/org/apache/kafka/image/NodeS3StreamSetObjectMetadataImage.java

```diff
@@ -36,6 +36,7 @@ public class NodeS3StreamSetObjectMetadataImage {
     private final int nodeId;
     private final long nodeEpoch;
     private final DeltaMap<Long/*objectId*/, S3StreamSetObject> s3Objects;
+    private final StreamOffsetIndexMap offsetIndexMap = new StreamOffsetIndexMap(2500000);
```

Comment:
The cache will be deprecated after generating a new image. So the cache may not be available after 1s.

Reply:
cache will be reconstructed regarding the latest order list, works fine on current scenario